joeynmt/training.py

```diff
@@ -425,12 +433,13 @@ def train_and_validate(self, train_data: Dataset, valid_data: Dataset) \
 
                     valid_duration = time.time() - valid_start_time
                     total_valid_duration += valid_duration
-                    self.logger.info(
-                        'Validation result (greedy) at epoch %3d, '
-                        'step %8d: %s: %6.2f, loss: %8.4f, ppl: %8.4f, '
-                        'duration: %.4fs', epoch_no + 1, self.steps,
-                        self.eval_metric, valid_score, valid_loss,
-                        valid_ppl, valid_duration)
+                    for idx, valid_score in enumerate(valid_scores):
+                        self.logger.info(
+                            'Validation result (greedy) at epoch %3d, '
+                            'step %8d: %s: %6.2f, loss: %8.4f, ppl: %8.4f, '
+                            'duration: %.4fs', epoch_no + 1, self.steps,
```

Comment:
Same here, just add all the metrics to the info string, and not repeat logging the stats.